### DISCUSSION TITLE
Great job : can we add the 880 BN Compass ?

### DISCUSSION THREAD

**Discussion by smichel19** [renatobo/bonogps#29] [Ideas]
Great dev .. thanks Renato.
I did the build with a 880 BN GPS, I just cannot get the compass work properly.
Do you think there is a way to have it working ? can I wire the SCL and SDA to the ESP32 ad try to read the values and alter the code to send/feed Harry Latimer with it ?
Apart from that it's great thanks !

**Comment by renatobo** [comment]
Glad to hear! iOS or Android?

HLT uses the magnetometer built into your phone so I let it do the dirty work to gain these benefits

- high frequency of data collection
- high precision potentially superior to the HMC5883L available on BN880
- no need to transmit its data (BLE is getting close to the limit in the case of iOS and text based NMEA) with a dedicated data structure because it does not merge with NMEA easily

Since HTL already has all of this and it does an excellent job, I never looked into using heading data from the BN880 - I don't even connect the wires.

There are ready to go libraries to connect to HMC5883L via i2c as

- https://www.arduino.cc/reference/en/libraries/grove-3-axis-digital-compass-hmc5883l/
- https://learn.adafruit.com/adafruit-hmc5883l-breakout-triple-axis-magnetometer-compass-sensor/wiring-and-test

The issue is that HLT is using your phone as the primary strategy so I don't even believe there is an option to do otherwise.

While not using the BN880 built-in magnetometer seems a waste of resources, I do not find benefits in having it done outside of the phone itself.
Harry also confirmed a similar opinion in other postings.

I'll be happy to hear other considerations on how to use it.

**Comment by smichel19** [comment]
I use IOS. Thanks for the info sharing. I understand the choice not to use it then.
One again Thanks for the dev... it is really nice ! 👍
